lib/rouge/lexers/cpp.rb

```diff
@@ -59,7 +59,7 @@ def self.reserved
       prepend :statements do
         rule %r/(class|struct)\b/, Keyword, :classname
         rule %r/template\b/, Keyword, :template
-        rule %r/\d+(\.\d+)?(?:h|(?:min)|s|(?:ms)|(?:us)|(?:ns))/, Num::Other
+        rule %r/#{dq}(\.#{dq})?(?:y|d|h|(?:min)|s|(?:ms)|(?:us)|(?:ns)|i|(?:if)|(?:il))\b/, Num::Other
```

Comment:
What do you think about moving the `i` to the end so we can remove the `\b` word boundary? 

```suggestion
        rule %r/#{dq}(\.#{dq})?(?:y|d|h|(?:min)|s|(?:ms)|(?:us)|(?:ns)|(?:if)|(?:il)|i)/, Num::Other
```

Reply:
I had thought about that solution, but liked the word boundary, as has been done for keywords in this file. My reasoning was that, for example, 1ns is a standard time literal but 1nsec is not. Leaving out the word boundary would allow it to be lexed as {Literal: 1ns}{Text: 'ec'}.

I'm not really sure the desired behavior here, so I'll leave it up to you as a maintainer. I don't have strong feelings either way.

Reply:
That makes sense. Let's leave as-is.